mlflow/pipelines/steps/predict.py

```diff
@@ -129,21 +129,36 @@ def _run(self, output_directory):
             if spark:
                 _logger.info("Found active spark session")
             else:
-                spark = _create_local_spark_session_for_pipelines()
                 _logger.info("Creating new spark session")
+                spark = _create_local_spark_session_for_pipelines()
         except Exception as e:
             raise MlflowException(
                 message=(
-                    "Encountered an error while searching for an active Spark session to"
-                    " score dataset with spark UDF. Please create a Spark session and try again."
+                    "Encountered an error while getting or creating an active Spark session to"
+                    " score dataset with spark UDF."
                 ),
                 error_code=BAD_REQUEST,
             ) from e
-        if not spark:
+
+        # check if output location is already populated for non-delta output formats
+        output_format = self.step_config["output_format"]
+        output_location = self.step_config["output_location"]
+        output_populated = False
+        if self.write_mode == "default":
+            if output_format == "parquet" or output_format == "delta":
```

Comment:
What's the difference between `delta` and `table`? Why is `delta` treated differently here?

Reply:
`delta` is also a delta table, but you specify the storage location / path for that delta table. With `table`, you don't specify the storage location (I believe it just gets written to the spark warehouse location) and it creates a delta table and adds it to the hive metastore under the specified table name. I'm open to removing combining these two options, now that we have a better understanding of how writing in delta works.